### ISSUE TITLE
SQLite backed cache: support listing of unwatchable resources

### ISSUE THREAD

**Issue by moio** [rancher/rancher#46768]
Parent epic: https://github.com/rancher/rancher/issues/46348

**Problem**: when the SQLite backed cache is enabled, Rancher uses Watch to keep the cache in sync. A few resource types are listable but not watchable, and that prevents legitimate list calls from succeeding.

Specifically, from a UI perspective, known regressions when the SQLite backed cache is enabled regard:
- the https://$RANCHER/dashboard/c/local/explorer/node page, which lists nodes in the local cluster and displays `nodemetrics` (`nodes metrics.k8s.io/v1beta1`). At the moment those are not displayed
- `componentstatuses cs` is deprecated in Kubernetes v1.19+ but is still discoverable via the magnifying glass search icon. At the moment nothing appears in the table
- `pods metrics.k8s.io/v1beta1` is discoverable via the magnifying glass search icon. At the moment nothing appears in the table
- any resource custom configured by users to be listable but not watchable

**Proposed solution**: teach Steve to add listable-but-not-watchable resources to the SQLite cache (`INSERT` on `list`). That will happen on-demand. Going through SQLite on each list operation is not ideal performance-wise, but it has the advantage of allowing us to reuse most logic (sorting/filtering/pagination)

**Expected outcome (UI perspective)**:
- list requests to nodemetrics, componentstatuses and podmetrics work regardless whether SQLite backed cache is enabled or not.

**More information**:
Related bug: https://github.com/rancher/rancher/issues/46030
Design discussion: https://suse.slack.com/archives/C06TMDMJQ0M/p1723650447631569

**Comment by moio** [comment]
CC @tomleb @richard-cox please let me know if I got anything wrong in the description

**Comment by moio** [comment]
@MKlimuszka per our meeting, this has lower priority than #46410, as the number of affected use cases is small, their importance is expected to be low and their frequency of usage is expected to be low as well.

**Comment by moio** [comment]
@tomleb I hope this mechanism can also be used for (imperative API) tokens

**Comment by tomleb** [comment]
Yeah I saw your message. It might work, I'll think more about it in a few weeks. :+1:

**Comment by brudnak** [comment]
Waiting on https://github.com/rancher/rancher/pull/49312 regression tests failing until this PR is merged

**Comment by ericpromislow** [comment]
What's the status on this now that the PR has been merged?

**Comment by brudnak** [comment]
Reopened due to https://github.com/rancher/rancher/issues/49415

**Comment by brudnak** [comment]
<details>
    <summary>🧪 Test Environment... CLICK TO EXPAND! ⬅️</summary>
<br>


| Component                        | Version / Type                      |
| -------------------------------- | ----------------------------------- |
| Rancher version                  | v2.11.0-alpha12                     |
| Installation option              | Helm (high availability)            |
| If Helm Chart k8s cluster        | v1.31.2                             |
| Cert Details                     | external tls `aws acm`              |
| Docker version                   | 20.10.7, build f0df350              |
| Downstream cluster type          | N/A                                 |
| Downstream K8s version           | N/A                                 |
| Authentication providers enabled | local                               |
| Logged in user role              | admin                               |
| Webhook version                  | rancher/rancher-webhook:v0.7.0-rc.9 |

</details>

<a name="top"></a>

### 🧪 Test Cases

| \#  | Priority | Description & Link                                 | PASS/FAIL | AUTO/MANUAL |
| --- | -------- | -------------------------------------------------- | --------- | ----------- |
| 1   | P0       | [Fresh Install DB Created (vai on)](#test-1)       | ✅ PASS    | 🤖 AUTOMATED |
| 2   | P0       | [Rancher Upgrade DB Create (vai on)](#test-2)      | ✅ PASS    | 🤖 AUTOMATED |
| 3   | P0       | [No Fail Watch Logs](#test-3)                      | ✅ PASS    | 🚧 MANUAL    |
| 4   | P0       | [(Regression Check) VAI Off Steve Filter](#test-4) | ✅ PASS    | 🤖 AUTOMATED |
| 5   | P0       | [(Regression Check) VAI Off Steve Limit](#test-5)  | ✅ PASS    | 🤖 AUTOMATED |
| 6   | P0       | [(Regression Check) VAI Off Steve Sort](#test-6)   | ✅ PASS    | 🤖 AUTOMATED |
| 7   | P0       | [(Regression Check) VAI On Steve Filter](#test-7)  | ✅ PASS    | 🤖 AUTOMATED |
| 8   | P0       | [(Regression Check) VAI On Steve Limit](#test-8)   | ✅ PASS    | 🤖 AUTOMATED |
| 9   | P0       | [(Regression Check) VAI On Steve Sort](#test-9)    | ✅ PASS    | 🤖 AUTOMATED |


<details>
    <summary>🚨 9 test cases... CLICK TO EXPAND! (For table links to work) ⬅️</summary>
<br>

# 1 / Fresh Install DB Created (vai on) Status: ✅ PASS

<a name="test-1"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 1 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

database tables to be present

**✅ Actual Outcome**

database tables were present

```log
=== RUN   TestVaiTestSuite
=== RUN   TestVaiTestSuite/TestVaiDisabled
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_with_negation_and_namespace
time="2025-03-12T10:30:47-05:00" level=info msg="Starting case: Filter with negation and namespace"
time="2025-03-12T10:30:47-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:47-05:00" level=info msg="Creating namespace: namespace-qduzxdne"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_with_negation_and_namespace (0.46s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_namespace
time="2025-03-12T10:30:48-05:00" level=info msg="Starting case: Filter by namespace"
time="2025-03-12T10:30:48-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:48-05:00" level=info msg="Creating namespace: namespace1-dfzsejwe"
time="2025-03-12T10:30:48-05:00" level=info msg="Creating namespace: namespace2-dfzsejwe"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_namespace (0.46s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_name_and_namespace
time="2025-03-12T10:30:48-05:00" level=info msg="Starting case: Filter by name and namespace"
time="2025-03-12T10:30:48-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:48-05:00" level=info msg="Creating namespace: namespace-rrhftwex"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_name_and_namespace (0.41s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_single_label
time="2025-03-12T10:30:49-05:00" level=info msg="Starting case: Filter by single label"
time="2025-03-12T10:30:49-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:49-05:00" level=info msg="Creating namespace: namespace-qlahgonk"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_single_label (0.40s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_multiple_labels
time="2025-03-12T10:30:49-05:00" level=info msg="Starting case: Filter by multiple labels"
time="2025-03-12T10:30:49-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:49-05:00" level=info msg="Creating namespace: namespace-ilozlpdr"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters/Filter_by_multiple_labels (0.51s)
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretFilters (2.24s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/PodFilters
=== RUN   TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_nginx_image
time="2025-03-12T10:30:50-05:00" level=info msg="Starting case: Filter by nginx image"
time="2025-03-12T10:30:50-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:50-05:00" level=info msg="Creating namespace: namespace1-tjracabj"
time="2025-03-12T10:30:50-05:00" level=info msg="Creating namespace: namespace2-tjracabj"
--- PASS: TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_nginx_image (0.58s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_busybox_image
time="2025-03-12T10:30:50-05:00" level=info msg="Starting case: Filter by busybox image"
time="2025-03-12T10:30:50-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:50-05:00" level=info msg="Creating namespace: namespace1-wyoedist"
time="2025-03-12T10:30:50-05:00" level=info msg="Creating namespace: namespace2-wyoedist"
--- PASS: TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_busybox_image (0.73s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_non-existent_image
time="2025-03-12T10:30:51-05:00" level=info msg="Starting case: Filter by non-existent image"
time="2025-03-12T10:30:51-05:00" level=info msg="Running with vai enabled: [false]"
time="2025-03-12T10:30:51-05:00" level=info msg="Creating namespace: namespace1-bmucufjh"
time="2025-03-12T10:30:51-05:00" level=info msg="Creating namespace: namespace2-bmucufjh"
--- PASS: TestVaiTestSuite/TestVaiDisabled/PodFilters/Filter_by_non-existent_image (0.86s)
--- PASS: TestVaiTestSuite/TestVaiDisabled/PodFilters (2.18s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_ascending
time="2025-03-12T10:30:52-05:00" level=info msg="Starting case: Sort by name ascending"
time="2025-03-12T10:30:52-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_ascending (0.64s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_descending
time="2025-03-12T10:30:52-05:00" level=info msg="Starting case: Sort by name descending"
time="2025-03-12T10:30:52-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_descending (0.69s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_ascending
time="2025-03-12T10:30:53-05:00" level=info msg="Starting case: Sort by namespace ascending"
time="2025-03-12T10:30:53-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_ascending (0.88s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_descending
time="2025-03-12T10:30:54-05:00" level=info msg="Starting case: Sort by namespace descending"
time="2025-03-12T10:30:54-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_descending (0.73s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_and_namespace_ascending
time="2025-03-12T10:30:55-05:00" level=info msg="Starting case: Sort by name and namespace ascending"
time="2025-03-12T10:30:55-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_and_namespace_ascending (0.71s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_and_name_ascending
time="2025-03-12T10:30:55-05:00" level=info msg="Starting case: Sort by namespace and name ascending"
time="2025-03-12T10:30:55-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_and_name_ascending (0.82s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_ascending_and_namespace_descending
time="2025-03-12T10:30:56-05:00" level=info msg="Starting case: Sort by name ascending and namespace descending"
time="2025-03-12T10:30:56-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_name_ascending_and_namespace_descending (0.68s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_descending_and_name_ascending
time="2025-03-12T10:30:57-05:00" level=info msg="Starting case: Sort by namespace descending and name ascending"
time="2025-03-12T10:30:57-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting/Sort_by_namespace_descending_and_name_ascending (0.60s)
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretSorting (5.74s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretLimit
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_50_secrets_with_limit_10
time="2025-03-12T10:30:57-05:00" level=info msg="Starting case: Paginate 50 secrets with limit 10"
time="2025-03-12T10:30:57-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_50_secrets_with_limit_10 (4.29s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_100_secrets_with_limit_25
time="2025-03-12T10:31:02-05:00" level=info msg="Starting case: Paginate 100 secrets with limit 25"
time="2025-03-12T10:31:02-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_100_secrets_with_limit_25 (9.05s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_30_secrets_with_limit_50
time="2025-03-12T10:31:11-05:00" level=info msg="Starting case: Paginate 30 secrets with limit 50"
time="2025-03-12T10:31:11-05:00" level=info msg="Running with vai enabled: [false]"
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretLimit/Paginate_30_secrets_with_limit_50 (2.41s)
--- PASS: TestVaiTestSuite/TestVaiDisabled/SecretLimit (15.75s)
=== RUN   TestVaiTestSuite/TestVaiDisabled/NormalOperations
--- PASS: TestVaiTestSuite/TestVaiDisabled/NormalOperations (0.19s)
--- PASS: TestVaiTestSuite/TestVaiDisabled (26.09s)
=== RUN   TestVaiTestSuite/TestVaiEnabled
time="2025-03-12T10:31:13-05:00" level=info msg="VAI is disabled, enabling it for test..."
=== RUN   TestVaiTestSuite/TestVaiEnabled/SetupVAIQuery
    vai_test.go:409: Building VAI query program on all pods...
    vai_test.go:417: Building VAI query on pod rancher-5ffc648559-26zc2
    vai_test.go:425: Executing command: kubectl exec rancher-5ffc648559-26zc2 -n cattle-system -- /bin/sh -c curl -k -sSL https://raw.githubusercontent.com/rancher/tests/refs/heads/main/validation/steve/vai/scripts/script.sh | sh
W0312 10:33:10.069404   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:33:10.264632   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:434: Successfully built VAI query on pod rancher-5ffc648559-26zc2
    vai_test.go:417: Building VAI query on pod rancher-5ffc648559-df46m
    vai_test.go:425: Executing command: kubectl exec rancher-5ffc648559-df46m -n cattle-system -- /bin/sh -c curl -k -sSL https://raw.githubusercontent.com/rancher/tests/refs/heads/main/validation/steve/vai/scripts/script.sh | sh
W0312 10:34:18.255935   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:34:18.428361   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:434: Successfully built VAI query on pod rancher-5ffc648559-df46m
    vai_test.go:417: Building VAI query on pod rancher-5ffc648559-x6dbd
    vai_test.go:425: Executing command: kubectl exec rancher-5ffc648559-x6dbd -n cattle-system -- /bin/sh -c curl -k -sSL https://raw.githubusercontent.com/rancher/tests/refs/heads/main/validation/steve/vai/scripts/script.sh | sh
W0312 10:35:22.861343   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:35:23.055969   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:434: Successfully built VAI query on pod rancher-5ffc648559-x6dbd
--- PASS: TestVaiTestSuite/TestVaiEnabled/SetupVAIQuery (204.49s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_with_negation_and_namespace
time="2025-03-12T10:35:25-05:00" level=info msg="Starting case: Filter with negation and namespace"
time="2025-03-12T10:35:25-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:25-05:00" level=info msg="Creating namespace: namespace-ofkijrcf"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_with_negation_and_namespace (1.47s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_namespace
time="2025-03-12T10:35:27-05:00" level=info msg="Starting case: Filter by namespace"
time="2025-03-12T10:35:27-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:27-05:00" level=info msg="Creating namespace: namespace1-hpjsftjq"
time="2025-03-12T10:35:27-05:00" level=info msg="Creating namespace: namespace2-hpjsftjq"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_namespace (0.93s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_name_and_namespace
time="2025-03-12T10:35:28-05:00" level=info msg="Starting case: Filter by name and namespace"
time="2025-03-12T10:35:28-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:28-05:00" level=info msg="Creating namespace: namespace-wtlnkbnr"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_name_and_namespace (0.27s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_single_label
time="2025-03-12T10:35:28-05:00" level=info msg="Starting case: Filter by single label"
time="2025-03-12T10:35:28-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:28-05:00" level=info msg="Creating namespace: namespace-bscwepgc"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_single_label (0.35s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_multiple_labels
time="2025-03-12T10:35:28-05:00" level=info msg="Starting case: Filter by multiple labels"
time="2025-03-12T10:35:28-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:28-05:00" level=info msg="Creating namespace: namespace-ogseezcx"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters/Filter_by_multiple_labels (0.38s)
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretFilters (3.40s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/PodFilters
=== RUN   TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_nginx_image
time="2025-03-12T10:35:29-05:00" level=info msg="Starting case: Filter by nginx image"
time="2025-03-12T10:35:29-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:29-05:00" level=info msg="Creating namespace: namespace1-wfiirayu"
time="2025-03-12T10:35:29-05:00" level=info msg="Creating namespace: namespace2-wfiirayu"
--- PASS: TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_nginx_image (0.74s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_busybox_image
time="2025-03-12T10:35:29-05:00" level=info msg="Starting case: Filter by busybox image"
time="2025-03-12T10:35:29-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:29-05:00" level=info msg="Creating namespace: namespace1-snzalwmg"
time="2025-03-12T10:35:29-05:00" level=info msg="Creating namespace: namespace2-snzalwmg"
--- PASS: TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_busybox_image (1.23s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_non-existent_image
time="2025-03-12T10:35:31-05:00" level=info msg="Starting case: Filter by non-existent image"
time="2025-03-12T10:35:31-05:00" level=info msg="Running with vai enabled: [true]"
time="2025-03-12T10:35:31-05:00" level=info msg="Creating namespace: namespace1-vexgwbnd"
time="2025-03-12T10:35:31-05:00" level=info msg="Creating namespace: namespace2-vexgwbnd"
--- PASS: TestVaiTestSuite/TestVaiEnabled/PodFilters/Filter_by_non-existent_image (0.84s)
--- PASS: TestVaiTestSuite/TestVaiEnabled/PodFilters (2.81s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_ascending
time="2025-03-12T10:35:31-05:00" level=info msg="Starting case: Sort by name ascending"
time="2025-03-12T10:35:31-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_ascending (0.74s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_descending
time="2025-03-12T10:35:32-05:00" level=info msg="Starting case: Sort by name descending"
time="2025-03-12T10:35:32-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_descending (0.90s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_ascending
time="2025-03-12T10:35:33-05:00" level=info msg="Starting case: Sort by namespace ascending"
time="2025-03-12T10:35:33-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_ascending (2.13s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_descending
time="2025-03-12T10:35:35-05:00" level=info msg="Starting case: Sort by namespace descending"
time="2025-03-12T10:35:35-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_descending (1.42s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_and_namespace_ascending
time="2025-03-12T10:35:37-05:00" level=info msg="Starting case: Sort by name and namespace ascending"
time="2025-03-12T10:35:37-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_and_namespace_ascending (0.82s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_and_name_ascending
time="2025-03-12T10:35:37-05:00" level=info msg="Starting case: Sort by namespace and name ascending"
time="2025-03-12T10:35:37-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_and_name_ascending (0.59s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_ascending_and_namespace_descending
time="2025-03-12T10:35:38-05:00" level=info msg="Starting case: Sort by name ascending and namespace descending"
time="2025-03-12T10:35:38-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_name_ascending_and_namespace_descending (0.64s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_descending_and_name_ascending
time="2025-03-12T10:35:39-05:00" level=info msg="Starting case: Sort by namespace descending and name ascending"
time="2025-03-12T10:35:39-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting/Sort_by_namespace_descending_and_name_ascending (0.60s)
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretSorting (7.84s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretLimit
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_50_secrets_with_limit_10
time="2025-03-12T10:35:39-05:00" level=info msg="Starting case: Paginate 50 secrets with limit 10"
time="2025-03-12T10:35:39-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_50_secrets_with_limit_10 (3.75s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_100_secrets_with_limit_25
time="2025-03-12T10:35:43-05:00" level=info msg="Starting case: Paginate 100 secrets with limit 25"
time="2025-03-12T10:35:43-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_100_secrets_with_limit_25 (7.20s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_30_secrets_with_limit_50
time="2025-03-12T10:35:50-05:00" level=info msg="Starting case: Paginate 30 secrets with limit 50"
time="2025-03-12T10:35:50-05:00" level=info msg="Running with vai enabled: [true]"
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretLimit/Paginate_30_secrets_with_limit_50 (2.35s)
--- PASS: TestVaiTestSuite/TestVaiEnabled/SecretLimit (13.31s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-26zc2
W0312 10:35:58.906108   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:35:59.075400   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-26zc2 (7.23s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-df46m
W0312 10:36:05.364152   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:05.518447   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-df46m (6.50s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-x6dbd
W0312 10:36:11.641682   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:11.812897   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:12.046946   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods/Checking_pod_rancher-5ffc648559-x6dbd (6.82s)
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckDBFilesInPods (20.66s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckSecretInDB
    vai_test.go:341: Starting checkSecretInVAIDatabase test
    vai_test.go:344: Generated secret name: db-secret-lbrlrxzo
    vai_test.go:352: Creating secret...
    vai_test.go:355: Secret created successfully
    vai_test.go:357: Listing Rancher pods...
    vai_test.go:360: Found 3 Rancher pods
    vai_test.go:362: Using script URL: https://raw.githubusercontent.com/rancher/tests/refs/heads/main/validation/steve/vai/scripts/script.sh
    vai_test.go:367: List all secrets to hydrate database...
    vai_test.go:372: Processing pod 1: rancher-5ffc648559-26zc2
    vai_test.go:379: Executing command on pod rancher-5ffc648559-26zc2
W0312 10:36:20.896365   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:21.074816   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:385: Command executed successfully on pod rancher-5ffc648559-26zc2
    vai_test.go:390: Secret found in pod rancher-5ffc648559-26zc2
    vai_test.go:398: Logging all outputs:
    vai_test.go:400: Output 1:
        Output from pod rancher-5ffc648559-26zc2:
        [2025-03-12 15:36:17] [INFO] Starting script execution...
        [2025-03-12 15:36:17] [INFO] Checking Go version:
        go version go1.23.6 linux/amd64
        [2025-03-12 15:36:17] [INFO] vai-query program already exists. Using existing binary.
        [2025-03-12 15:36:17] [INFO] Executing the query program...
        2025/03/12 15:36:17.454465 Starting VAI database query...
        2025/03/12 15:36:17.455110 Opening connection to original database...
        2025/03/12 15:36:17.455615 Creating database snapshot...
        2025/03/12 15:36:18.526613 Opening connection to snapshot database...
        2025/03/12 15:36:18.526783 Querying table '_v1_Secret_fields' for resource 'db-secret-lbrlrxzo'
        2025/03/12 15:36:18.528224 Found resource 'db-secret-lbrlrxzo' in table '_v1_Secret_fields'
        2025/03/12 15:36:18.528256 Cleaning up snapshot...
        Found resource: db-secret-lbrlrxzo
        2025/03/12 15:36:18.528292 Query operation completed
        [2025-03-12 15:36:18] [INFO] Script execution completed.
    vai_test.go:403: Secret found status: true
    vai_test.go:405: checkSecretInVAIDatabase test completed
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckSecretInDB (8.77s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/CheckNamespaceInAllVAIDatabases
    vai_test.go:459: Starting TestCheckNamespaceInAllVAIDatabases test
    vai_test.go:463: Generated namespace name: db-namespace-zorcyqsk
    vai_test.go:470: Creating namespace...
    vai_test.go:473: Namespace created successfully
    vai_test.go:483: Found 3 Rancher pods
    vai_test.go:491: Processing pod 1: rancher-5ffc648559-26zc2
    vai_test.go:442: Checking namespace db-namespace-zorcyqsk in pod rancher-5ffc648559-26zc2
W0312 10:36:29.483462   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:29.656466   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:499: Namespace found in pod rancher-5ffc648559-26zc2
    vai_test.go:491: Processing pod 2: rancher-5ffc648559-df46m
    vai_test.go:442: Checking namespace db-namespace-zorcyqsk in pod rancher-5ffc648559-df46m
W0312 10:36:36.764548   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:37.119668   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:499: Namespace found in pod rancher-5ffc648559-df46m
    vai_test.go:491: Processing pod 3: rancher-5ffc648559-x6dbd
    vai_test.go:442: Checking namespace db-namespace-zorcyqsk in pod rancher-5ffc648559-x6dbd
W0312 10:36:45.079819   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:45.233766   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:499: Namespace found in pod rancher-5ffc648559-x6dbd
    vai_test.go:507: Logging all outputs:
    vai_test.go:509: Output 1:
        Output from pod rancher-5ffc648559-26zc2:
        2025/03/12 15:36:25.306513 Starting VAI database query...
        2025/03/12 15:36:25.306716 Opening connection to original database...
        2025/03/12 15:36:25.306760 Creating database snapshot...
        2025/03/12 15:36:26.386621 Opening connection to snapshot database...
        2025/03/12 15:36:26.386705 Querying table '_v1_Namespace_fields' for resource 'db-namespace-zorcyqsk'
        2025/03/12 15:36:26.388140 Found resource 'db-namespace-zorcyqsk' in table '_v1_Namespace_fields'
        2025/03/12 15:36:26.388182 Cleaning up snapshot...
        2025/03/12 15:36:26.388216 Query operation completed
        Found resource: db-namespace-zorcyqsk
    vai_test.go:509: Output 2:
        Output from pod rancher-5ffc648559-df46m:
        2025/03/12 15:36:33.144337 Starting VAI database query...
        2025/03/12 15:36:33.144488 Opening connection to original database...
        2025/03/12 15:36:33.144644 Creating database snapshot...
        Found resource: db-namespace-zorcyqsk
        2025/03/12 15:36:34.198978 Opening connection to snapshot database...
        2025/03/12 15:36:34.199033 Querying table '_v1_Namespace_fields' for resource 'db-namespace-zorcyqsk'
        2025/03/12 15:36:34.200283 Found resource 'db-namespace-zorcyqsk' in table '_v1_Namespace_fields'
        2025/03/12 15:36:34.200297 Cleaning up snapshot...
        2025/03/12 15:36:34.200325 Query operation completed
    vai_test.go:509: Output 3:
        Output from pod rancher-5ffc648559-x6dbd:
        2025/03/12 15:36:41.428157 Starting VAI database query...
        2025/03/12 15:36:41.428346 Opening connection to original database...
        2025/03/12 15:36:41.428421 Creating database snapshot...
        2025/03/12 15:36:42.481044 Opening connection to snapshot database...
        2025/03/12 15:36:42.481124 Querying table '_v1_Namespace_fields' for resource 'db-namespace-zorcyqsk'
        2025/03/12 15:36:42.482750 Found resource 'db-namespace-zorcyqsk' in table '_v1_Namespace_fields'
        2025/03/12 15:36:42.482774 Cleaning up snapshot...
        Found resource: db-namespace-zorcyqsk
        2025/03/12 15:36:42.482810 Query operation completed
    vai_test.go:512: Namespace found count: 3
    vai_test.go:516: TestCheckNamespaceInAllVAIDatabases test completed
--- PASS: TestVaiTestSuite/TestVaiEnabled/CheckNamespaceInAllVAIDatabases (23.92s)
=== RUN   TestVaiTestSuite/TestVaiEnabled/checkMetricTablesInVAIDatabase
    vai_test.go:521: Starting checkMetricTablesInVAIDatabase test
    vai_test.go:604: Accessing metrics endpoint to ensure tables are created
    vai_test.go:609: Listing metrics.k8s.io.nodemetrics resources
    vai_test.go:615: Successfully accessed metrics endpoint
    vai_test.go:531: Listing Rancher pods...
    vai_test.go:534: Found 3 Rancher pods
    vai_test.go:536: Using script URL: https://raw.githubusercontent.com/rancher/tests/refs/heads/main/validation/steve/vai/scripts/script.sh
    vai_test.go:547: Processing pod 1: rancher-5ffc648559-26zc2
    vai_test.go:555: Executing command on pod rancher-5ffc648559-26zc2
W0312 10:36:52.553066   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:36:52.707899   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:562: Command executed successfully on pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: go version go1.23.6 linux/amd64 in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:48.816712 Starting VAI database query... in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:48.816894 Opening connection to original database... in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:48.816916 Creating database snapshot... in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:49.872518 Opening connection to snapshot database... in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:49.872695 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices") in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:49.874992 Cleaning up snapshot... in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: 2025/03/12 15:36:49.875037 Query operation completed in pod rancher-5ffc648559-26zc2
    vai_test.go:578: Found table: No results found in pod rancher-5ffc648559-26zc2
    vai_test.go:547: Processing pod 2: rancher-5ffc648559-df46m
    vai_test.go:555: Executing command on pod rancher-5ffc648559-df46m
W0312 10:36:59.907108   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:37:00.108279   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:562: Command executed successfully on pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: go version go1.23.6 linux/amd64 in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:56.396419 Starting VAI database query... in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:56.396559 Opening connection to original database... in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:56.396580 Creating database snapshot... in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:57.458835 Opening connection to snapshot database... in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:57.458899 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices") in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:57.460290 Cleaning up snapshot... in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: 2025/03/12 15:36:57.460318 Query operation completed in pod rancher-5ffc648559-df46m
    vai_test.go:578: Found table: No results found in pod rancher-5ffc648559-df46m
    vai_test.go:547: Processing pod 3: rancher-5ffc648559-x6dbd
    vai_test.go:555: Executing command on pod rancher-5ffc648559-x6dbd
W0312 10:37:09.448616   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0312 10:37:09.605151   31264 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
    vai_test.go:562: Command executed successfully on pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: go version go1.23.6 linux/amd64 in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:05.188072 Starting VAI database query... in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:05.188880 Opening connection to original database... in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:05.189317 Creating database snapshot... in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:06.277799 Opening connection to snapshot database... in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:06.277882 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices") in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:06.280119 Cleaning up snapshot... in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: 2025/03/12 15:37:06.280196 Query operation completed in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: name in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: ---- in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: metrics.k8s.io_v1beta1_NodeMetrics in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: metrics.k8s.io_v1beta1_NodeMetrics_indices in pod rancher-5ffc648559-x6dbd
    vai_test.go:578: Found table: metrics.k8s.io_v1beta1_NodeMetrics_fields in pod rancher-5ffc648559-x6dbd
    vai_test.go:585: Logging all outputs:
    vai_test.go:587: Output 1:
        Output from pod rancher-5ffc648559-26zc2:
        [2025-03-12 15:36:48] [INFO] Starting script execution...
        [2025-03-12 15:36:48] [INFO] Checking Go version:
        go version go1.23.6 linux/amd64
        [2025-03-12 15:36:48] [INFO] vai-query program already exists. Using existing binary.
        [2025-03-12 15:36:48] [INFO] Executing the query program...
        2025/03/12 15:36:48.816712 Starting VAI database query...
        2025/03/12 15:36:48.816894 Opening connection to original database...
        2025/03/12 15:36:48.816916 Creating database snapshot...
        2025/03/12 15:36:49.872518 Opening connection to snapshot database...
        2025/03/12 15:36:49.872695 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices")
        2025/03/12 15:36:49.874992 Cleaning up snapshot...
        2025/03/12 15:36:49.875037 Query operation completed
        No results found
        [2025-03-12 15:36:49] [INFO] Script execution completed.
    vai_test.go:587: Output 2:
        Output from pod rancher-5ffc648559-df46m:
        [2025-03-12 15:36:56] [INFO] Starting script execution...
        [2025-03-12 15:36:56] [INFO] Checking Go version:
        go version go1.23.6 linux/amd64
        [2025-03-12 15:36:56] [INFO] vai-query program already exists. Using existing binary.
        [2025-03-12 15:36:56] [INFO] Executing the query program...
        2025/03/12 15:36:56.396419 Starting VAI database query...
        2025/03/12 15:36:56.396559 Opening connection to original database...
        2025/03/12 15:36:56.396580 Creating database snapshot...
        2025/03/12 15:36:57.458835 Opening connection to snapshot database...
        2025/03/12 15:36:57.458899 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices")
        2025/03/12 15:36:57.460290 Cleaning up snapshot...
        2025/03/12 15:36:57.460318 Query operation completed
        No results found
        [2025-03-12 15:36:57] [INFO] Script execution completed.
    vai_test.go:587: Output 3:
        Output from pod rancher-5ffc648559-x6dbd:
        [2025-03-12 15:37:05] [INFO] Starting script execution...
        [2025-03-12 15:37:05] [INFO] Checking Go version:
        go version go1.23.6 linux/amd64
        [2025-03-12 15:37:05] [INFO] vai-query program already exists. Using existing binary.
        [2025-03-12 15:37:05] [INFO] Executing the query program...
        2025/03/12 15:37:05.188072 Starting VAI database query...
        2025/03/12 15:37:05.188880 Opening connection to original database...
        2025/03/12 15:37:05.189317 Creating database snapshot...
        2025/03/12 15:37:06.277799 Opening connection to snapshot database...
        2025/03/12 15:37:06.277882 Executing custom SQL query: SELECT name FROM sqlite_master WHERE type="table" AND name IN ("metrics.k8s.io_v1beta1_NodeMetrics", "metrics.k8s.io_v1beta1_NodeMetrics_fields", "metrics.k8s.io_v1beta1_NodeMetrics_indices")
        2025/03/12 15:37:06.280119 Cleaning up snapshot...
        2025/03/12 15:37:06.280196 Query operation completed
        name
        ----
        metrics.k8s.io_v1beta1_NodeMetrics
        metrics.k8s.io_v1beta1_NodeMetrics_indices
        metrics.k8s.io_v1beta1_NodeMetrics_fields
        
        3 row(s) returned
        [2025-03-12 15:37:06] [INFO] Script execution completed.
    vai_test.go:600: checkMetricTablesInVAIDatabase test completed
--- PASS: TestVaiTestSuite/TestVaiEnabled/checkMetricTablesInVAIDatabase (24.42s)
--- PASS: TestVaiTestSuite/TestVaiEnabled (356.89s)
--- PASS: TestVaiTestSuite (484.90s)
PASS

Tests passed: 58 0f 58 tests

```

</details>
<hr>

# 2 / Rancher Upgrade DB Create (vai on) Status: ✅ PASS

<a name="test-2"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 2 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

database tables to be present

**✅ Actual Outcome**

database tables were present


</details>
<hr>

# 3 / No Fail Watch Logs Status: ✅ PASS

<a name="test-3"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 3 details... Click to expand</summary>

**Test Steps for Validation**

1. Start rancher/rancher on version with the fix
2. Enable the ui-sql-cache feature flag
3. Go to https://$RANCHER/v1/metrics.k8s.io.nodemetrics in your browser


**✅ Expected Outcome**

No watch log errors

**✅ Actual Outcome**

No watch log errors present


</details>
<hr>

# 4 / (Regression Check) VAI Off Steve Filter Status: ✅ PASS

<a name="test-4"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 4 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should filter when vai is off

**✅ Actual Outcome**

Steve successfully filtered with vai off 


</details>
<hr>

# 5 / (Regression Check) VAI Off Steve Limit Status: ✅ PASS

<a name="test-5"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 5 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should limit when vai is off

**✅ Actual Outcome**

Steve successfully used limit with vai off 


</details>
<hr>



# 6 / (Regression Check) VAI Off Steve Sort Status: ✅ PASS

<a name="test-6"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 6 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should sort when vai is off

**✅ Actual Outcome**

Steve successfully sorted with vai off 


</details>
<hr>

# 7 / (Regression Check) VAI On Steve Filter Status: ✅ PASS

<a name="test-7"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 7 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should filter when vai is on

**✅ Actual Outcome**

Steve successfully filtered with vai on 


</details>
<hr>

# 8 / (Regression Check) VAI On Steve Limit Status: ✅ PASS

<a name="test-8"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 8 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should limit when vai is on

**✅ Actual Outcome**

Steve successfully used limit with vai on 


</details>
<hr>

# 9 / (Regression Check) VAI On Steve Sort Status: ✅ PASS

<a name="test-9"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 9 details... Click to expand</summary>

**Test Steps for Validation**

1. Automated check

**✅ Expected Outcome**

Steve should sort when vai is on

**✅ Actual Outcome**

Steve successfully sorted with vai on 


</details>
<hr>

</details>
